sawmill-core/src/main/java/io/logz/sawmill/processors/DocSignatureProcessor.java

```diff
@@ -0,0 +1,136 @@
+package io.logz.sawmill.processors;
+
+import io.logz.sawmill.Doc;
+import io.logz.sawmill.ProcessResult;
+import io.logz.sawmill.Processor;
+import io.logz.sawmill.annotations.ProcessorProvider;
+import io.logz.sawmill.exceptions.ProcessorExecutionException;
+import io.logz.sawmill.utilities.JsonUtils;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@ProcessorProvider(type = "docSignature", factory = DocSignatureProcessor.Factory.class)
+public class DocSignatureProcessor implements Processor {
+    private final SignatureMode signatureMode;
+    private final Set<String> includeValueFields;
+    private final String DOC_SIGNATURE_FIELD = "logzio_doc_signature";
+    public DocSignatureProcessor(SignatureMode signatureMode, Set<String> includeValueFields) {
+        this.signatureMode = signatureMode;
+        this.includeValueFields = includeValueFields;
+    }
+
+    @Override
+    public ProcessResult process(Doc doc) throws InterruptedException {
+        String signature;
+        try {
+            signature = createSignature(doc);
+        } catch (Exception e) {
+            return ProcessResult.failure(
+                    "failed to create signature, SignatureMode: " + signatureMode,
+                    new ProcessorExecutionException(DocSignatureProcessor.class.getSimpleName(), e));
+        }
+
+        if(signature.isEmpty())
+            return ProcessResult.failure("signature is empty, SignatureMode: " + signatureMode);
```

Comment:
consider

Reply:
i think we should keep it.. consider a scenario where the algorithm is broken but does not throw an exception...